src/lib/mode-watcher.svelte

```diff
@@ -18,23 +27,23 @@
 		};
 	});
 
-	function setInitialMode(
-		defaultMode: 'light' | 'dark' | 'system',
-		themeColors?: { dark: string; light: string }
-	) {
-		const elem = document.documentElement,
-			mode = localStorage.getItem('mode') || defaultMode,
-			light =
-				mode === 'light' ||
-				(mode === 'system' && window.matchMedia('(prefers-color-scheme: light)').matches);
+	function setInitialMode(defaultMode: Mode, themeColors?: ThemeColors) {
+		const rootEl = document.documentElement;
+		const mode = localStorage.getItem('mode') || defaultMode;
+		const light =
+			mode === 'light' ||
+			(mode === 'system' && window.matchMedia('(prefers-color-scheme: light)').matches);
 
-		elem.classList[light ? 'remove' : 'add']('dark');
-		elem.style.colorScheme = light ? 'light' : 'dark';
+		rootEl.classList[light ? 'remove' : 'add']('dark');
```

Comment:
`elem` and `te` was chosen instead of more descriptive names for bundle size reasons. but maybe this was not needed, maybe this code is also minimized. will have to double check!

Reply:
Yeah man these get minified. Descriptive names > short names (unless you're building a minifier 😉)

Reply:
I could have sworn that this kind of hacky code where you include something that's stringified as a raw html string would not be minified but I am probably mistaken

Reply:
if you can find `rootEl` in the client-side bundle I'll send you some $ 😂

Reply:
haha sure 🤣

does this count?

<img width="1512" alt="Screenshot 2024-01-25 at 11 54 20" src="https://github.com/svecosystem/mode-watcher/assets/31876997/89ea3a63-e6e4-433f-ab14-b6d268017023">

this is from running `pnpm build && pnpm preview`. maybe this is not a legit test though?

other functions ARE minimized here but it seems like the FOUC snippet is still untouched with `rootEl` and the other variable names intact

Reply:
compare this to the `mode.<hash>.js` file which is clearly minimized:
<img width="1512" alt="Screenshot 2024-01-25 at 11 58 05" src="https://github.com/svecosystem/mode-watcher/assets/31876997/1cfb84ce-398e-4189-a358-86b5e8763893">

Reply:
and here is the older version of `mode-watcher` running in production when the root element was still called `elem` instead of `rootEl`:
<img width="1512" alt="Screenshot 2024-01-25 at 12 00 42" src="https://github.com/svecosystem/mode-watcher/assets/31876997/74440629-ab44-4139-872d-520839557df9">

Reply:
it will not be a big difference - but short variable names (duh) and lowercase letters (better compression IIRC) will lead to less bytes being sent to the end user of every app that uses `mode-watcher` unless I am missing something @huntabyte

Reply:
Oh, wtf it's because we're placing that directly in like that 🤔🤔. This can definitely be minified/shorter than that, let me investigate this a bit. If we look at https://skeleton.dev, we'll see they have similar functionality but only include this bit in their client-side bundle, which is how I thought we were doing it:
![CleanShot 2024-01-25 at 10 21 47@2x](https://github.com/svecosystem/mode-watcher/assets/64506580/dd9f9ddf-0769-4ae8-993d-af392c92205f)

Regardless, you were correct so I have sent some 💵 to your Ko-fi 😂

Reply:
hmm, I wonder why our method is not minified and their version is 🤔 I wonder if this broke at some point when I refactored it so that it can take a dynamic default mode/theme color settings.

I can also take a stab at investigating this if you get stuck, let me know!

> Regardless, you were correct so I have sent some 💵 to your Ko-fi 😂

haha, you didn't have to do that, but thanks! let me know if you want me to return it 😆

Reply:
No you're fine, I should have done a more thorough review before making bold claims like that 😂 It's because we are stringifying it, and _then_ slapping that stringified version directly into the tag, vs calling the `setInitialMode.toString()` in there.

Reply:
I thought so too, so I tried:
```svelte
	{@html `<\u{73}cript nonce="%sveltekit.nonce%">(${setInitialMode.toString()})("dark");</script>`}
```
similar to skeleton, but that did not work it seems

Reply:
maybe, the function needs to be defined in a `.ts` file and exported, let me try that

Reply:
tried:
```svelte
<script lang="ts">
	import { setInitialMode } from './mode.js';
</script>
...
// mode-watcher.svelte
<svelte:head>
	{@html `<\u{73}cript nonce="%sveltekit.nonce%">(${setInitialMode.toString()})("dark");</script>`}
</svelte:head>
```

with:
```
// mode.ts
export function setInitialMode(defaultMode: Mode, themeColors?: ThemeColors) {
	const rootEl = document.documentElement;
	const mode = localStorage.getItem('mode-watcher-mode') || defaultMode;
	const light =
		mode === 'light' ||
		(mode === 'system' && window.matchMedia('(prefers-color-scheme: light)').matches);

	rootEl.classList[light ? 'remove' : 'add']('dark');
	rootEl.style.colorScheme = light ? 'light' : 'dark';

	if (themeColors) {
		const themeMetaEl = document.querySelector('meta[name="theme-color"]');
		if (themeMetaEl) {
			themeMetaEl.setAttribute('content', mode === 'light' ? themeColors.light : themeColors.dark);
		}
	}

	localStorage.setItem('mode', mode);
}
```

and that still does not work...

Reply:
I don't get it.

I copied the exact code from skeleton now, just to verify:

```svelte
<svelte:head>
	<!-- Workaround for a svelte parsing error: https://github.com/sveltejs/eslint-plugin-svelte/issues/492 -->
	{@html `<\u{73}cript nonce="%sveltekit.nonce%">(${setInitialClassState.toString()})();</script>`}
</svelte:head>
```

and

```ts
// mode.ts
export function setInitialClassState() {
	const elemHtmlClasses = document.documentElement.classList;
	// Conditions
	const condLocalStorageUserPrefs = localStorage.getItem('modeUserPrefers') === 'false';
	const condLocalStorageUserPrefsExists = !('modeUserPrefers' in localStorage);
	const condMatchMedia = window.matchMedia('(prefers-color-scheme: dark)').matches;
	// Add/remove `.dark` class to HTML element
	if (condLocalStorageUserPrefs || (condLocalStorageUserPrefsExists && condMatchMedia)) {
		elemHtmlClasses.add('dark');
	} else {
		elemHtmlClasses.remove('dark');
	}
}
```

still not minified when running `pnpm build && pnpm preview`. I have no idea what's going on